packages/babel-plugin-proposal-decorators/src/transformer.js

```diff
@@ -7,8 +7,11 @@ function prop(key, value) {
   return t.objectProperty(t.identifier(key), value);
 }
 
-function value(body, params = []) {
-  return t.objectMethod("method", t.identifier("value"), params, body);
+function value(body, params = [], async, generator) {
+  const method = t.objectMethod("method", t.identifier("value"), params, body);
```

Comment:
I guess this is because we don't require you do to async/generator? A weird bug just since we don't pass it down but not sure what we would want to do in general for it

Reply:
`t.functionDeclaration` accepts async/generator but `t.objectMethod` doesn't.
I'd love to have a function to specify the fields I want, like
```js
t.objectMethod.with({
  kind: "method",
  key: t.identifier("value"),
  params, body, async, generator
});
```